components/feature/search/src/main/java/mozilla/components/feature/search/SearchUseCases.kt

```diff
@@ -97,6 +99,6 @@ class SearchUseCases(
     }
 
     val newTabSearch: NewTabSearchUseCase by lazy {
-        NewTabSearchUseCase(context, searchEngineManager, sessionManager)
+        NewTabSearchUseCase(context, searchEngineManager, sessionManager, isPrivate)
     }
```

Comment:
In the issue you mention `NewPrivateTabSearchUseCase`. But it looks like instead of adding a new property here, you decided to add a `private´ parameter. Does this work better in your situation?

Reply:
I realized that the NewPrivateTabSearchUseCase would be the exact same excpet for having an isPrivate flag. To avoid code duplication, I think it makes more sense to just instantiate a type of NewTabSearchUseCase that "isPrivate." Let me know if you disagree!

Reply:
I wonder if we should just introduce another property though, like

```kotlin
    val newPrivateTabSearch by lazy {
        NewTabSearchUseCase(context, searchEngineManager, sessionManager)
    }
```

... which avoids having multiple `SearchUseCases` instances?

Reply:
I wonder if we should just introduce another property though, like

```kotlin
    val newPrivateTabSearch by lazy {
        NewTabSearchUseCase(context, searchEngineManager, sessionManager)
    }
```

... which avoids having multiple `SearchUseCases` instances?

Reply:
Ah, I see. That's a much better solution!

Reply:
I've updated the PR to reflect this. Is this more what you had in mind?

Reply:
If we want a separate use case, another option would be to mark `NewTabSearchUseCase` (and its invoke method) as `open` and subclass it:

```
 class NewPrivateTabSearchUseCase(
        context: Context,
        searchEngineManager: SearchEngineManager,
        sessionManager: SessionManager
    ) : NewTabSearchUseCase(context, searchEngineManager, sessionManager) {

        override fun invoke(searchTerms: String, source: Session.Source, selected: Boolean, private: Boolean) {
            super.invoke(searchTerms, source, selected, true)
        }
    }
```

Reply:
Doesn't make any difference the caller though as they'll just use the `newPrivateTabSearch` field. So, matter of preference mostly.